src/lib/sqlite/turso.ts

```diff
@@ -25,3 +60,354 @@ export async function migrateRemote(
+      migrationStateStore.set('done')
+    } catch (e: unknown) {
+      console.error(e)
+      migrationStateStore.set('error')
+    }
+  })()
+
+  let snippetsStore: Reloadable<Snippet[]>
+  if (db === null) {
+    snippetsStore = asyncDerived(
+      [],
+      async () => {
+        return []
+      },
+      {
+        reloadable: true,
+      }
+    ) as Reloadable<Snippet[]>
+  } else {
+    // @ts-ignore
+    snippetsStore = createSnippetsStore(db, folderId)
+  }
+  await snippetsStore.load()
+
+  return {
+    subscribe: snippetsStore.subscribe,
+    async clone(snippet: Snippet): Promise<void> {
+      if (db === null) {
+        return
+      }
+
+      const clonedSnippet: Snippet = {
+        ...snippet,
+        id: crypto.randomUUID(),
+        position: get(snippetsStore).length + 1,
+        tags: snippet.tags.slice(),
+        createdAt: new Date().getTime(),
+        updatedAt: new Date().getTime(),
+      }
+      const dbSnippet = displaySnippetToDbSnippet(folderId, clonedSnippet)
+      await upsertSnippet(db, dbSnippet)
+      if (snippet.tags.length > 0) {
+        await upsertTags(db, clonedSnippet.id, snippet.tags)
+      }
+
+      await snippetsStore.reload()
+    },
+    async upsert(snippet: Snippet): Promise<void> {
+      if (db === null) {
+        return
+      }
+
+      const dbSnippet = displaySnippetToDbSnippet(folderId, snippet)
+      await upsertSnippet(db, dbSnippet)
+      if (snippet.tags && snippet.tags.length > 0) {
+        await clearTags(db, snippet.id)
+        await upsertTags(db, snippet.id, snippet.tags)
+      }
+      await snippetsStore.reload()
+    },
+    async remove(id: string): Promise<void> {
+      if (db === null) {
+        return
+      }
+
+      await deleteSnippet_(db, id)
+      await snippetsStore.reload()
+    },
+    async move(
+      movingSnippet: Snippet,
+      sourceFolderId: string,
+      destinationFolderId: string,
+    ) {
+      if (db === null) {
+        return
+      }
+
+      const dbSnippet = displaySnippetToDbSnippet(sourceFolderId, movingSnippet)
+      dbSnippet.folderId = destinationFolderId
+      await upsertSnippet(db, dbSnippet)
+
+      await snippetsStore.reload()
+    },
+    async clear() {
+      if (db === null) {
+        return
+      }
+      await deleteSnippetsByFolder(db, folderId)
+      await snippetsStore.reload()
+    },
+    async clearAll() {
+      if (db === null) {
+        return
+      }
+      await deleteAllSnippets(db)
+      await snippetsStore.reload()
+    },
+    async clearAllTags() {
+      if (db === null) {
+        return
+      }
+      await clearAllTags_(db)
+      await snippetsStore.reload()
+    },
+    async isAvailable(): Promise<boolean> {
+      return db !== null;
+    },
+    async refresh() {
+      await snippetsStore.reload()
+    },
+    migrationStateStore,
+  }
+}
+
+export async function createRemoteSnippetsStoreV2(
+  dbPairStore: Readable<DbPair>,
+  folderIdStore: Readable<string>,
```

Comment:
There was a strange bug: pressing "Merge" for conflicted records wouldn't work as expected.